esrally/client/asynchronous.py

```diff
@@ -77,7 +77,7 @@ async def send(self, conn: "Connection") -> "ClientResponse":
         self.response = self.response_class(
             self.method,
             self.original_url,
-            writer=self._writer,
+            writer=self._writer,  # type: ignore[arg-type]  # TODO remove this ignore when introducing type hints
```

Comment:
We could also just an an `assert self._writer is not None` above this line, and it won't complain. 

Oh, wait... It might not be necessary depending on the `response_class` we've got, huh?

Reply:
I haven't analyzed this deeply and just went for a safe type ignore, but thinking about this now wouldn't such assertion fail at runtime? (I haven't tested)

My understanding is `StaticRequest` class extends `aiohttp.ClientRequest` to support static API responses feature of Rally which allows us to test Rally itself providing curated responses from non-existing ES (ES delays and performance eliminated from the equation). The extension is done by overwriting `send()` method. However, `_writer` property is updated in the _original_ `send()` method. Otherwise it defaults to `None` because `_writer` property is taken from `__writer` class variable which is set to `None` initially.

Reply:
Ah, that is frustrating. Eh, we can get back to it later, then